lib/srv/desktop/windows_server.go

```diff
@@ -863,8 +853,25 @@ func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger,
 		AllowDirectorySharing: authCtx.Checker.DesktopDirectorySharing(),
 		ShowDesktopWallpaper:  s.cfg.ShowDesktopWallpaper,
 	})
+	// before we check the error above, we grab the windows user so that we
+	// can build our audit event generator
+	var windowsUser string
```

Comment:
nit: this is such an edge case that it arguably doesn't matter, and still ugly, but we could do something like
```suggestion
	windowsUser := "not received"
```

and change `New` to 

```go
// New creates and connects a new Client based on cfg.
func New(cfg Config) (*Client, error) {
	if err := cfg.checkAndSetDefaults(); err != nil {
		return nil, err
	}
	c := &Client{
		cfg:           cfg,
		readyForInput: 0,
	}

	if err := c.readClientUsername(); err != nil {
		return nil, trace.Wrap(err)
	}
	if err := cfg.AuthorizeFn(c.username); err != nil {
		return nil, trace.Wrap(err)
	}
	if err := c.readClientSize(); err != nil {
		return nil, trace.Wrap(err)
	}
	return c, nil
}
```

to maximize our chances of getting a name in the `rdpc.GetClientUsername()` below as well as make more clear what the currently empty name means

Reply:
I'd rather keep the name empty if we don't have it. (This is how all of our audit events work)